### PR TITLE
[KV] Overhaul KV API docs 

### PR DESCRIPTION
**Pull request by thomasgauvin**
#15814 since refactor

### CODE REVIEW

**Comment by thomasgauvin** [comment]
@kodster28 can we fix the ai linter so it doesn't throw on "namespace"? KV namespaces is the official term

**Comment by thomasgauvin** [comment]
@kodster28 why is ai linter saying wrangler should not be capitalized? It is capitalized everywhere else in docs and is a product https://developers.cloudflare.com/workers/wrangler/

**Comment by bllchmbrs** [comment]
@thomasgauvin , I can help! (dev @ hyperlint)! I'll give you some instructions momentarily to quiet some of these. All alerts are controllable by you all, so this is all pretty natural for just getting the style guide introduced as you build up the vocabulary. Give me one moment and I'll edit this and provide some links!

---
edit:

**Step 1**
[edit this file](https://github.com/cloudflare/cloudflare-docs/blob/thomasgauvin-kv-commands-overhaul-2/.github/styles/config/vocabularies/cloudflare/accept.txt)

Things to do in that file...
- **Wrangler** - Right now, it has wrangler lower-cased and upper cased in the vocabulary So it doesn't know what should be done.
- add the words `namespace` and `serializable` and `boolean`, `lookups`

Once you've done that, you can just ignore `Resolve Conversation` on those issues.

**Comment by bllchmbrs** [comment]
@thomasgauvin , see comment above. Feel free to ping me with any other questions!

edit: btw hyperlint runs in apart on workers and uses KV. Love the product!

**Comment by thomasgauvin** [comment]
@bllchmbrs thanks, I'll make sure these are added to our dictionary! I didn't expect y'all to hop on this thread, thanks again! 🙌

**Review by Oxyjun — APPROVED** [review]

File: src/content/docs/kv/api/delete-key-value-pairs.mdx [hunk 1]
```diff
@@ -2,20 +2,75 @@
...
+To delete a key-value pair, call the `delete()` method of the [KV binding](/kv/concepts/kv-bindings/) on any KV namespace you have bound to your Worker code:
+
+```js
+env.NAMESPACE.delete(key);
+```
+
+#### Parameters
+
+* `key`: `string`
+  * The key to associate with the value.
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/delete-key-value-pairs.mdx:55]:
```suggestion
- `key`: `string`
  - The key to associate with the value.
```

File: src/content/docs/kv/api/delete-key-value-pairs.mdx [hunk 2]
```diff
@@ -2,20 +2,75 @@
...
+
+#### Response
+
+* `response`: `Promise<void>`
+  * A `Promise` that resolves if the delete is successful.
+
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/delete-key-value-pairs.mdx:61]:
```suggestion
- `response`: `Promise<void>`
  - A `Promise` that resolves if the delete is successful.

```

File: src/content/docs/kv/api/read-key-value-pairs.mdx [hunk 3]
```diff
@@ -2,95 +2,170 @@
...
+The `get()` method returns a promise you can `await` on to get the value. If the key is not found, the promise will resolve with the literal value `null`.
 
-* `text`: A `string` (default).
-* `json`: An object decoded from a JSON string.
-* `arrayBuffer`: An [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) instance.
-* `stream`: A [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream).
+#### Parameters
 
-For simple values, use the default `text` type which provides you with your value as a `string`. For convenience, a `json` type is also specified which will convert a JSON value into an object before returning the object to you. For large values, use `stream` to request a `ReadableStream`. For binary values, use `arrayBuffer` to request an `ArrayBuffer`.
+* `key` `string`
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/read-key-value-pairs.mdx]:
```suggestion
* `key`: `string`
```

File: src/content/docs/kv/api/read-key-value-pairs.mdx [hunk 4]
```diff
@@ -2,95 +2,170 @@
...
+  * The key of the KV pair.
+* `type` `"text" | "json" | "arrayBuffer" | "stream"`
+  * Optional. The type of the value to be returned. `string` is the default.
+* `options` `{
+    cacheTtl: number,
+    type: "text" | "json" | "arrayBuffer" | "stream"
+}`
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/read-key-value-pairs.mdx]:
```suggestion
  * The key of the KV pair.
* `type`: `"text" | "json" | "arrayBuffer" | "stream"`
  * Optional. The type of the value to be returned. `string` is the default.
* `options`: `{
    cacheTtl: number,
    type: "text" | "json" | "arrayBuffer" | "stream"
}`
```

File: src/content/docs/kv/api/read-key-value-pairs.mdx [hunk 5]
```diff
@@ -2,95 +2,170 @@
...
+  * Optional. Object containing the `cacheTtl` and `type` properties. The `cacheTtl` property defines the length of time in seconds that a KV result is cached in the global network location it is accessed from (minimum: 60). The `type` property defines the type of the value to be returned. 
 
-For large values, the choice of `type` can have a noticeable effect on latency and CPU usage. For reference, the `type` can be ordered from fastest to slowest as `stream`, `arrayBuffer`, `text`, and `json`.
+#### Response
 
-## CacheTtl parameter
+* `response` `Promise<string | Object | ArrayBuffer | ReadableStream | null>`
```
Comment (thomasgauvin) [inline src/content/docs/kv/api/read-key-value-pairs.mdx]:
```suggestion
* `response`: `Promise<string | Object | ArrayBuffer | ReadableStream | null>`
```

**Pull request merged**